activesupport/lib/active_support/concern.rb

```diff
@@ -104,6 +104,12 @@ def initialize
       end
     end
 
+    class UnreachableIncludedBlock < StandardError #:nodoc:
+      def initialize
+        super "Cannot add an 'included' block after Concern was already included"
```

Comment:
I think we should explain exactly what's going on in this error message, if I got into this situation as a developer, this error would just confuse me more. Instead of focusing on the "illegal" condition, let's just say that the included block cannot be executed, and why that's happening.